torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -4421,9 +4454,10 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::gather(
 
   // avoidRecordStreams_ note: collective() will stash inputTensors and
   // outputs, which == outputTensors[0] on the root rank where it matters.
+
   return collective(
-      inputTensor,
-      outputs[0], // just to fit the collective interface
+      inputTensors,
```

Comment:
nit: I think it would be clearer to create a vector to "just to fit the collective interface" than directly reusing `inputTensors` here.
`inputTensors` carries the notion of multiple devices, but that's not what the `collective` API means here.